lib/astc_encode.cpp

```diff
@@ -803,3 +884,232 @@ ktxTexture2_CompressAstc(ktxTexture2* This, ktx_uint32_t quality) {
 
     return ktxTexture2_CompressAstcEx(This, &params);
 }
+
+struct decompression_workload
+{
+    astcenc_context* context;
+    uint8_t* data;
+    size_t data_len;
+    astcenc_image* image_out;
+    astcenc_swizzle swizzle;
+    astcenc_error error;
+};
+
+/**
+ * @brief Runner callback function for a decompression worker thread.
+ *
+ * @param thread_count   The number of threads in the worker pool.
+ * @param thread_id      The index of this thread in the worker pool.
+ * @param payload        The parameters for this thread.
+ */
+static void decompression_workload_runner(int thread_count, int thread_id, void* payload) {
+    (void)thread_count;
+
+    decompression_workload* work = static_cast<decompression_workload*>(payload);
+    astcenc_error error = astcenc_decompress_image(work->context, work->data, work->data_len,
+                                                   work->image_out, &work->swizzle, thread_id);
+    // This is a racy update, so which error gets returned is a random, but it
+    // will reliably report an error if an error occurs
+    if (error != ASTCENC_SUCCESS)
+    {
+        work->error = error;
+    }
+}
+
+/**
+ * @brief Decodes the provided ktx2 texture if its astc encoded
+ *
+ * @param This     The texture to decode
+ * @param vkformat The decoding format to use
+ */
+// will update "This" with the uncompressed copy
+KTX_API KTX_error_code KTX_APIENTRY
+ktxTexture2_DecodeAstc(ktxTexture2 *This, ktx_uint32_t vkformat) {
```

Comment:
@MarkCallow In-case it didn't register, as you can see I have introduced this "Decode" method inside `ktx.h` and `astc_encode.cpp` this might sound a bit weird but I didn't think creating a seprate `astc_decode.cpp` file was worth it. If you don't agree please shout.

Reply:
It did register but thanks for the added notice. Does the decoder share much code with what is already in `astc_encode.cpp`? If it does not it is probably better to create `astc_decode.cpp`. Maybe we can change `command_extract.cpp` to use it. On second thoughts maybe not. It only needs to decode the image(s) being extracted not the whole texture.

Reply:
It does reuse bits from `astc_encode.cpp` thats mainly why I put it there and some more decode specific bits I added too but more importantly it will be using bits from `astc_encode.cpp` for HDR stuff as well going forward.

Reply:
Okay. Leave it in the same file. Let's rename the file to `astc_codec.cpp`.